react_juce/core/EcmascriptEngine_Hermes.cpp

```diff
@@ -0,0 +1,522 @@
+        jsi::Object varObjectToJSIObject(juce::DynamicObject *v, jsi::Runtime &runtime)
+        {
+            jassert(v);
+
+            jsi::Object jsiObj(runtime);
+            for (auto &prop : v->getProperties())
+            {
+                jsiObj.setProperty(runtime, prop.name.getCharPointer(), varToJSIValue(prop.value, runtime));
+            }
+
+            return jsiObj;
+        }
+
+        jsi::Function varMethodToJSIMethod(juce::var::NativeFunction &v, jsi::Runtime &runtime)
+        {
+            return jsi::Function::createFromHostFunction(
+                runtime,
+                jsi::PropNameID::forAscii(runtime, ""),
+                0,
+                [v = v](jsi::Runtime& rt, const jsi::Value& thisVal, const jsi::Value* args, size_t count)
+                {
+                    std::vector<juce::var> varArgs;
+                    for (size_t i = 0; i < count; ++i)
+                    {
+                        varArgs.push_back(jsiValueToVar(args[i], rt));
+                    }
+
+                    juce::var::NativeFunctionArgs nfArgs(jsiValueToVar(thisVal, rt), varArgs.data(), static_cast<int>(count));
+                    return varToJSIValue(v(nfArgs), rt);
+                }
+            );
+        }
+
+        jsi::Array varArrayToJSIArray(const juce::Array<juce::var> *v, jsi::Runtime &runtime)
+        {
+            jassert(v);
+
+            const auto numItems = static_cast<size_t>(v->size());
+            jsi::Array jsiArray(runtime, numItems);
+
+            for (size_t i = 0; i < numItems; ++i)
+            {
+                jsiArray.setValueAtIndex(runtime, i, varToJSIValue(v->getReference(i), runtime));
+            }
+
+            return jsiArray;
+        }
+
+        jsi::Value varToJSIValue(const juce::var &v, jsi::Runtime &runtime)
+        {
+            if (v.isBool())
+                return jsi::Value(static_cast<bool>(v));
+
+            if (v.isInt())
+                return jsi::Value(static_cast<int>(v));
+
+            if (v.isDouble())
+                return jsi::Value(static_cast<double>(v));
+
+            if (v.isString())
+                return jsi::Value(runtime,  jsi::String::createFromAscii(runtime, v.toString().getCharPointer()));
+
+            if (v.isUndefined())
+                return jsi::Value::undefined();
+
+            if (v.isVoid())
+                return {};
+
+            if (v.isArray())
+                return varArrayToJSIArray(v.getArray(), runtime);
+
+            if (v.isMethod())
+            {
+                auto nf = v.getNativeFunction();
+                return varMethodToJSIMethod(nf, runtime);
+            }
+
+            if (v.isObject())
+                return varObjectToJSIObject(v.getDynamicObject(), runtime);
+
+            if (v.isBinaryData())
+            {
+                //TODO: Currently appears no way to create a jsi::ArrayBuffer from C++.
```

Comment:
This is annoying. I'll speak to the Hermes maintainers about this once I've fired other contributions back but in principle I don't think it's likely to be an issue for 99% of users right now.

Reply:
Yea, no worries. I don't think anyone will use this yet